anchore_engine/db/entities/policy_engine.py

```diff
@@ -3033,3 +3058,9 @@ def select_nvd_classes(db=None):
 
     log.debug("selected {}/{} nvd classes".format(_nvd_cls, _cpe_cls))
     return _nvd_cls, _cpe_cls
+
+
+def _search_dict(key, val, collection):
+    for entry in collection:
+        if entry[key] == val:
+            return entry
```

Comment:
```suggestion
def _get_one_or_none(key, val, collection):
    """
    Find a match for the object in the collection using the key-value pair. Return the result only if 1 match is found

    Example instance of collection
    [
        {
          "source": "CVE ID",
          "url": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=2000-0089"
        },
        {
          "source": "Bugtraq ID",
          "url": "http://www.securityfocus.com/bid/947"
        },
    ]
    """
    if not key or not val:
        return None

    result = None
    for entry in collection:
        if entry.get(key) == val:  # use get operator here instead of [] to avoid KeyError
            if result:
                return None  # > 1 result found, bail out of here and return None
            else:
                result = entry

    return result
```

Reply:
alternatively:
```
filtered = [ x for x in collection where x.get(key) == val]
if len(filtered) != 1:
  return None
else:
 return filtered[0]
```

Reply:
Thanks for the pointers.